src/app/users/create-user/create-user.component.ts

```diff
@@ -1,56 +1,137 @@
-import { Component, OnInit, ViewChild } from '@angular/core';
-import { Router } from '@angular/router';
-import { FormioComponent } from '@formio/angular';
+import { AfterContentChecked, AfterViewChecked, AfterViewInit, Component, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Formio, FormioComponent, FormioUtils } from '@formio/angular';
 import { UsersService } from 'src/app/services/users.service';
 import { IUserItem } from '../user-item-model';
 import { MdbModalRef } from 'mdb-angular-ui-kit/modal';
 import { MasterForms } from 'src/app/services/utility/master.forms.constants';
 import { RoleService } from 'src/app/services/roles.service';
 import { combineLatest, map } from 'rxjs';
 import { DepartMentService } from 'src/app/services/departments.service';
+import { formatDate } from '@angular/common';
+import { DateUtil } from 'src/app/services/utility/DateUtil';
 
 @Component({
   selector: 'app-create-user',
   templateUrl: './create-user.component.html',
   styleUrls: ['./create-user.component.scss']
 })
 export class CreateUserComponent implements OnInit {
+  title: string = 'Create'
   isDataLoaded: boolean = false;
+
   @ViewChild(FormioComponent, { static: false })
   form!: FormioComponent;
+  username: string = '';
+  user!: IUserItem;
+  roles!: any[];
+  departments!: any[];
+
   FormData = MasterForms.CREATE_USER_FORM_TEMPLATE;
   constructor(private userService: UsersService,
     private roleService: RoleService,
     private departMentService: DepartMentService,
     private router: Router,
-  ) { }
+    private activatedRoute: ActivatedRoute,
+  ) {
+    const navigation: any = this.router.getCurrentNavigation();
+    if (navigation.extras.state) {
```

Comment:
This won't work if we refresh the page. We won't have any data on navigation state then.

Reply:
Correct, but in that case I am fetching data using apis, check on line:58